src/main/java/ch/njol/skript/variables/Variables.java

```diff
@@ -272,15 +273,19 @@ static Lock getReadLock() {
 	 */
 	@Nullable
 	public final static Object getVariable(final String name, final @Nullable Event e, final boolean local) {
-		if (local) {
+		String n = name;
+        if (SkriptConfig.caseInsensitiveVariables.value()) {

```

Comment:
Querying config options can be potentially expensive, so it should not be done in a method that is called very often. Suggestion: use changer in the option to set a static field somewhere, then check against it.

Reply:
I don't think that'd be the case as config values are only parsed on startup & `/sk reload config/all` but if you think so, I'll make this change myself.